src/action_handle_hashtags.py

```diff
@@ -0,0 +1,143 @@
+                   on_interaction):
+    interaction = partial(_interact_with_user,
+                          likes_count=likes_count,
+                          follow_percentage=follow_percentage,
+                          on_like=on_like,
+                          profile_filter=profile_filter)
+
+    if not _open_post_hashtags(device, hashtag):
+        return
+    _iterate_over_posts(device, interaction, storage, on_interaction)
+
+def _open_post_hashtags(device, hashtag):
+    if hashtag is None:
+        print("Enter own hashtag search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+    else:
+        print("Press search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+        search_button.click.wait()
+
+        print("Open hashtag #" + hashtag)
+        search_edit_text = device(resourceId='com.instagram.android:id/action_bar_search_edit_text',
+                                  className='android.widget.EditText')
+        search_edit_text.set_text(hashtag)
+
+        search_hashtag_button = device(resourceId='com.instagram.android:id/fixed_tabbar_view',
+                            className='android.widget.FrameLayout').child(index=2)
+        search_hashtag_button.click.wait()
+
+        device.wait.idle()
+        hashtag_view = device(resourceId='com.instagram.android:id/row_hashtag_textview_tag_name',
+                               className='android.widget.TextView',
+                               text="#" + hashtag)
+
+        if not hashtag_view.exists:
+            print_timeless(COLOR_FAIL + "Cannot find hashtag #" + hashtag + ", abort." + COLOR_ENDC)
+            return False
+
+        hashtag_view.click.wait()
+
+        print("Open #" + hashtag + " recent posts")
+        hashtag_type = device(resourceId='com.instagram.android:id/tab_layout',
+                                  className='android.widget.LinearLayout').child(index=1)
+        hashtag_type.click.wait()
+
+    return True
+
+def open_photo(device,index_ran):
+        # recycler_view has a className 'androidx.recyclerview.widget.RecyclerView' on modern Android versions and
+        # 'android.view.View' on Android 5.0.1 and probably earlier versions
+        recycler_view = device(resourceId='com.instagram.android:id/recycler_view',
+                                className='androidx.recyclerview.widget.RecyclerView')
+        item_view = recycler_view.child(index=index_ran)
+        item_view.click.wait()
+
+
+def _iterate_over_posts(device, interaction, storage, on_interaction):
+    #random up to 12 because min show on screen intially is 12 the max is 20

```

Comment:
So this feature is limited to interact with only 12 first users, isn't it?

That's going to finish too quickly then. I see 2 ways how to improve:
1. Add scrolling to the current solution
2. OR go another way: don't open users' profiles. Instead open first and only first post by a hashtag -> like it -> scroll down randomly -> like again -> scroll down randomly -> like again -> ... etc.

What do you think?

Reply:
No, it does scroll at the bottom of the screen, but in the function _iterate_over_posts.  Reason for 12 is on the intial screen theres is only 12 posts max then when you scroll theres 18.

Reply:
Oh yeah sorry